Improved scim setup instructions (#1430)
oktadeveloper/okta-blog: _source/_posts/2023-07-28-scim-workshop.md

## Add a SCIM file and create a scimRoute

> For maintainability, we'll aim to keep our SCIM implementation code in one place. Let’s create a SCIM file, scim.ts, in `apps/api/src`, and import it in `main.ts`. At the top of `apps/api/src/scim.ts`, you'll need to import Router to create a scimRoute to export to `apps/api/src/main.ts`. Copy paste the following code to the top of your apps/api/src/scim.ts file:

Author: ```suggestion
For maintainability, we'll aim to keep our SCIM implementation code in one place. Let's create a SCIM file, scim.ts, in `apps/api/src`, and import it in `main.ts`. At the top of `apps/api/src/scim.ts`, you'll need to import Router to create a scimRoute to export to `apps/api/src/main.ts`. Copy paste the following code to the top of your apps/api/src/scim.ts file:
```

```ts
import { Router } from 'express';
export const scimRoute = Router();
```

> And you’ll then import the scimRoute in `apps/api/src/main.ts` and after `import session from 'express-session';` as shown below:

Author: ```suggestion
And you'll then import the scimRoute in `apps/api/src/main.ts` and after `import session from 'express-session';` as shown below:
```

import passportLocal from 'passport-local';
import passportOIDC from 'passport-openidconnect';
import passport from 'passport';
import session from 'express-session';
import passportBearer from 'passport-http-bearer';

// body-parser is required to accept the header content-type application/scim+json from Okta
// https://www.npmjs.com/package/body-parser
// RFC Notes: https://datatracker.ietf.org/doc/html/rfc7644#section-3.1
import bodyParser from 'body-parser';
```
And instruct the app to use `bodyParser` for the relevant content types below. Be sure to add this to your SCIM related code section in `apps/api/src/main.ts` as well. It should look like the following:  

```ts
///////////////////////////////////////////////////////
// SCIM-related routes

app.use(bodyParser.json({ type: 'application/scim+json' }));

// '/scim/v2' path appends to every SCIM Endpoints 
// Okta recommended url - https://developer.okta.com/docs/guides/scim-provisioning-integration-prepare/main/#base-url
app.use('/scim/v2', scimRoute);

const port = process.env.PORT || 3333;
const server = app.listen(port, () => {
  console.log(`Listening at http://localhost:${port}/api`);
});

> server.on('error', console.error);

Author: ```suggestion
server.on('error', console.error);
(```)
### Add HTTP Logging
```

import passportOIDC from 'passport-openidconnect';
import passport from 'passport';
import session from 'express-session';
import passportBearer from 'passport-http-bearer';

// body-parser is required to accept the header content-type application/scim+json from Okta
// https://www.npmjs.com/package/body-parser
// RFC Notes: https://datatracker.ietf.org/doc/html/rfc7644#section-3.1
import bodyParser from 'body-parser';
```
And instruct the app to use `bodyParser` for the relevant content types below. Be sure to add this to your SCIM related code section in `apps/api/src/main.ts` as well. It should look like the following:  

```ts
///////////////////////////////////////////////////////
// SCIM-related routes

app.use(bodyParser.json({ type: 'application/scim+json' }));

// '/scim/v2' path appends to every SCIM Endpoints 
// Okta recommended url - https://developer.okta.com/docs/guides/scim-provisioning-integration-prepare/main/#base-url
app.use('/scim/v2', scimRoute);

const port = process.env.PORT || 3333;
const server = app.listen(port, () => {
  console.log(`Listening at http://localhost:${port}/api`);
});
server.on('error', console.error);

> ### Add HTTP Logging

Author: ```suggestion
```

### Add HTTP Logging
```

import passportOIDC from 'passport-openidconnect';
import passport from 'passport';
import session from 'express-session';
import passportBearer from 'passport-http-bearer';

// body-parser is required to accept the header content-type application/scim+json from Okta
// https://www.npmjs.com/package/body-parser
// RFC Notes: https://datatracker.ietf.org/doc/html/rfc7644#section-3.1
import bodyParser from 'body-parser';
```
And instruct the app to use `bodyParser` for the relevant content types below. Be sure to add this to your SCIM related code section in `apps/api/src/main.ts` as well. It should look like the following:  

```ts
///////////////////////////////////////////////////////
// SCIM-related routes

app.use(bodyParser.json({ type: 'application/scim+json' }));

// '/scim/v2' path appends to every SCIM Endpoints 
// Okta recommended url - https://developer.okta.com/docs/guides/scim-provisioning-integration-prepare/main/#base-url
app.use('/scim/v2', scimRoute);

const port = process.env.PORT || 3333;
const server = app.listen(port, () => {
  console.log(`Listening at http://localhost:${port}/api`);
});
server.on('error', console.error);

> ```

Author: ```suggestion
```

```

> ### Add HTTP Logging

Reviewer: I think you need a newline above?

> 

Author: ```suggestion
```

### Add HTTP Logging
```
